src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexPersistence.cs

```diff
@@ -166,7 +170,9 @@ IndexSearcher CreateIndexSearcher(IState state)
                             {
                                 var newReader = reader.Reopen(state);
                                 if (newReader != reader)
-                                    reader.DecRef(state);
+                                {
+                                    reader.Dispose();
```

Comment:
How do you know that the old reader isn't still in use?

Reply:
I was looking at the dispose: 
https://github.com/ravendb/lucenenet/blob/3506f5028832315a4d6f8aeababcdd96b87bbac1/src/Lucene.Net/Index/IndexReader.cs#L1145

Which seems to be a "single-shot", so if we dispose twice and the ref count drop to 1 only on the second time, we will not do the actual dispose.

which make me think that this is wrong ?
https://github.com/ravendb/ravendb/blob/3e8d206e3dc7a9e611a1f253c68df6d1883ea613/src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexPersistence.cs#L205

Reply:
What happens when you have a _query_ that is running while this is also running.
The `readerLock` you have isn't covered by those, and you may dispose the reader while queries are also running, no?
That is why we have the `DecRef` there.